### ISSUE TITLE
Apparent conflict between `-mtune=native` and `-march=native`

### ISSUE THREAD

**Issue by fxcoudert** [iains/gcc-14-branch#2]
Reported first at https://github.com/fxcoudert/gfortran-for-macOS/issues/65

```
meau /tmp $ gfortran-14 a.f90 -O3 -mtune=native -march=native             
f951: Error: unknown value 'apple-m1' for '-march'
f951: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a armv8.6-a armv8.7-a armv8.8-a armv8.9-a armv8-r armv9-a armv9.1-a armv9.2-a armv9.3-a armv9.4-a native
f951: note: did you mean '-mcpu=apple-m1'?
```

**Comment by iains** [comment]
yeah.. I think that (like PowerPC) the incantation is -mcpu= for AArch64

**Comment by iains** [comment]
although, having checked with one of our Arm colleagues, apparently both should work (mcpu is said to be more user-friendly) .. so we need to look at why -march is being missed.

I'm currently chasing the second bootstrap break for Darwin [this week] so it will not be immediate.

**Comment by iains** [comment]
A better explanation from said colleague:

using e.g. -march=armv8.2-a+sve will give you "architectural" SVE codegen whereas e.g. -mcpu=neoverse-v1 will give SVE codegen that's better tuned to that core.

So, for our users (who are not, generally, equipped with suitable info to do -march=armv8.4-a+xx+yy+zz), we should be strongly recommending -mcpu=apple-mN.

**Comment by iains** [comment]
to allow -march=native to work, if that is a useful thing to do, I guess that means converting the apple-mN into the equivalent "armv8.x-a+...." string.

$ ./gcc/xgcc -Bgcc /source/test/hello.c -mcpu=native -mtune=native 
works OK AFAICT.

**Comment by barracuda156** [comment]
> yeah.. I think that (like PowerPC) the incantation is -mcpu= for AArch64

What is the reason behind `-march` not being recognized for Power* btw? It breaks quite a lot of software, since developers unfamiliar with this peculiarity just assume that it works for any architecture, and otherwise requires rather ugly fixes with special case for Power* changing the flag to `-mtune` and/or `-mcpu`.
Admittedly, I know nothing about the internals of these optimizations, but it seems reasonable at least to make an alias from `-march` to `-mcpu` for Power*, if `-march` as a distinct option cannot be supported or makes little sense to support.
